task8-main/src/main/java/com/example/classloader/app/Demo.java

```diff
@@ -0,0 +1,82 @@
+package com.example.classloader.app;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.*;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.util.*;
+
+public class Demo {
+    private static final Logger LOGGER = LoggerFactory.getLogger(Demo.class);
+
+    public static void main(String[] args) throws IOException, ClassNotFoundException, NoSuchMethodException, InstantiationException, IllegalAccessException, InvocationTargetException {
+
+        String pathName = getModulsLocation("location.properties");
+        List<File> jars = Arrays.asList(new File(pathName).listFiles());
+        URL[] urls = new URL[jars.size()];
+        for (int i = 0; i < jars.size(); i++) {
+            try {
+                urls[i] = jars.get(i).toURI().toURL();
+                System.out.println(urls[i]);
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
+        }
+        URLClassLoader childClassLoader = new URLClassLoader(urls, ClassLoader.getSystemClassLoader());
+        Thread.currentThread().setContextClassLoader(childClassLoader);
+
+        LOGGER.info("Enter the operation name for calculation (Add/Multiply/Square Root): ");
+        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
+        String operation = reader.readLine();
+
+        LOGGER.info("Enter the parameters to calculate the result of operation (press any character to exit): ");
+        Scanner in = new Scanner(System.in);
+
+        ArrayList<Double> paramList = new ArrayList<>();
+        while (in.hasNextDouble()) {
+            double param = in.nextDouble();
+            paramList.add(param);
+        }
+        LOGGER.info(paramList.toString());
+
+        double result = performCalculation(childClassLoader, operation, paramList);
+        LOGGER.info("The result is " + result);
+    }
+
+    private static double performCalculation(URLClassLoader childClassLoader, String operationName, ArrayList<Double> paramList) throws ClassNotFoundException, InvocationTargetException, InstantiationException, IllegalAccessException, NoSuchMethodException {
+
+        switch (operationName.toLowerCase()) {
+            case "add":
+                return getCalculationResult(childClassLoader, "com.example.classloader.implement.CalculateAdd", paramList);
```

Comment:
you need to load all classes from specified folder's jar files and for Calculable classes you may create custom `Map<String, Class<? extends Calculable>>` with list of implementation classes